ocs_ci/helpers/helpers.py

```diff
@@ -835,6 +835,30 @@ def validate_cephfilesystem(fs_name):
     return True if (ceph_validate and ocp_validate) else False
 
 
+def delete_all_resource_of_kind_containing_string(kind, search_string):
+    """
+    Delete all resource of kind specified with name containing search_string.
+    Args:
+        kind (str): kind of the resource to search and delete.
+        search_string (str): Search string for the resources to be searched and deleted.
+    Returns
+        bool: True if deletion of all resource succeeded else false.
+
+    """
+    resource_dict_list = get_all_resource_of_kind_containing_string(kind=kind, search_string=search_string)
+    if len(resource_dict_list) > 0:
+        for resource in resource_dict_list:
+            OCS(**resource).delete()
+        resource_dict_after_list = get_all_resource_of_kind_containing_string(kind=kind, search_string=search_string)
+        if len(resource_dict_after_list) > 0:
+            for resource in resource_dict_after_list:
+                logger.error(f"Resource {resource.resource_name} was not deleted successfully")
+            return False
+        else:
+            return True
+    return True
```

Comment:
if `len(resource_dict_list)` is 0 we return True. Is this intentional?

Reply:
As you can see `get_all_resource_of_kind_containing_string` is happening twice - one for getting the object and deleting them and the second one is to verify that there are no leftovers, so if there are no leftovers `len(resource_dict_list) is 0` I'm returning true.

Reply:
ack